x-pack/plugins/cross_cluster_replication/public/app/components/follower_index_form.js

```diff
@@ -123,6 +151,10 @@ export const FollowerIndexForm = injectI18n(
       return this.state.followerIndex;
     };
 
+    toggleAdvancedSettings = () => {
+      this.setState(({ areAdvancedSettingsVisible }) => ({ areAdvancedSettingsVisible: !areAdvancedSettingsVisible }));
```

Comment:
When the settings are hidden, we need to clear the relevant validation and fields state, so that any old input for this section is safely ignored. If we want to make this a _really_ nice UX, we can store the previous input in a cache, and reapply the cached values to the fields when the use shows the advanced settings again. This way the user won't lose any input by hiding the advanced settings.

Without these changes, invalid input still prevents saving the follower index, even though the advanced settings are hidden (and I believe valid input would be included in the save request).

![image](https://user-images.githubusercontent.com/1238659/51156639-592b8800-1831-11e9-9d77-6954a142dea7.png)

Reply:
I think this can be partially resolved with your ["Use advanced settings" suggestion](https://github.com/elastic/kibana/pull/28267/files/5fdd0536332fb4a771640ff3089693506deb2695#r247752584)